### DISCUSSION TITLE
Merge a single pull request on an exact date/time

### DISCUSSION THREAD

**Discussion by emmahsax** [community/community#26958] [Actions]
<p>Hello, I’m new to GitHub Actions. I’m looking for the best way to merge a single pull request on an exact date/time that I set up for <em>that specific</em> pull request (so not on a general cron schedule). Then, I would like a bot or GitHub Actions to merge that pull request for me.</p>
<p>I did find this GitHub Action, and I tested it out: <a href="https://github.com/marketplace/actions/merge-schedule" rel="nofollow noopener">https://github.com/marketplace/actions/merge-schedule</a>. But it ended up just not working how I would expect, and I went to look for a different solution.</p>
<p>So instead I’ve found <a href="https://github.com/marketplace/actions/merge-pull-requests" rel="nofollow noopener">https://github.com/marketplace/actions/merge-pull-requests</a>, which has more options, which I like, and I’ve fully tested it and it does work just how I’d expect.</p>
<p>I set this new GitHub Action up as running on a schedule, and here’s my workflow for my action: <a href="https://github.com/emma-sax4/emma-sax4.github.io/blob/6d1810b306b203ce20fcb1500df1c76fc2c5e643/.github/workflows/auto_merge_pull_requests.yml" rel="nofollow noopener">https://github.com/emma-sax4/emma-sax4.github.io/blob/6d1810b306b203ce20fcb1500df1c76fc2c5e643/.github/workflows/auto_merge_pull_requests.yml</a>.</p>
<p>But every now and then, I have a pull request that I want to merge on a schedule that *doesn’t match* what’s currently in my default branch.</p>
<p>If I change the cron’s schedule in my pull request, will the action then run and execute on the newly defined schedule? As an example, my hope is that a pull request like this (<a href="https://github.com/emma-sax4/emma-sax4.github.io/pull/215/files" rel="nofollow noopener">https://github.com/emma-sax4/emma-sax4.github.io/pull/215/files</a>) would trigger the GitHub Action to run at 22:40 UTC on Fridays, which would merge its own pull request.</p>
<p>Or do I need to change/add the schedule to my default branch first for the action to run at 22:40 UTC?</p>
<p>Or, perhaps there’s an even better way to do what I’m trying, or a new GitHub Action I haven’t thought of.</p>
<p>Thanks in advance.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/emma-sax4">@emma-sax4</a> ,</p>
<p>I noticed that there are some actions in the <a href="https://github.com/marketplace?query=merge+PR&amp;type=actions" rel="nofollow noopener"><strong>GitHub Marketplace</strong></a> can auto-merge the PR with specified lables. You can try to set two cron for the schedule event in the workflow,</p>
<p>for example,</p>
<pre><code>schedule:
  - cron: '20 21 * * 5'
  - cron: '40 22 * * 5'
</code></pre>
<p>then set different jobs run on different crons to merge PRs with different labels,</p>
<pre><code>jobs:
  job1:
    if: github.event.schedule == '20 21 * * 5'
. . .
steps:
Auto-merge PR with label-1

job2:
if: github.event.schedule == '40 22 * * 5'
. . .
steps:
Auto-merge PR with label-2
</code></pre>
<p>In addition, I also find an APP ( <strong><a href="https://github.com/marketplace/pr-scheduler" rel="nofollow noopener">PR Scheduler</a> </strong>) from the marketplace, I think it should be available to your case.</p>

**Comment by emmahsax** [comment, marked as answer]
<p>The <a href="https://github.com/marketplace/pr-scheduler" rel="nofollow noopener">PR scheduler app</a> actually worked like a charm! It was exactly what I wanted!</p>
<p>Thank you!</p>

**Reply by woeterman94** [reply]
4 merges per month. How generous.

**Reply by emmahsax** [reply]
I didn't realize the limit was so low. I only do maybe 1-2 per month, so I don't think I've ever hit that. Or perhaps the app has made a limit when there wasn't always a limit.
